### PR TITLE
Create Google mock for ReadOnlyVariable2

### PR DESCRIPTION
**Pull request by VivianNK**
## Description

Created a new PR for cleaner single-commit (also this one works)
Old PR: https://github.com/microsoft/mu_basecore/pull/530

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Tested with ConfigKnobShimLib GoogleTest

## Integration Instructions

N/A

### CODE REVIEW

File: MdePkg/Test/Mock/Include/GoogleTest/Ppi/MockReadOnlyVariable2.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     IN OUT    CHAR16                          *VariableName,
+     IN OUT    EFI_GUID                        *VariableGuid)
+    );
+};
+
+extern "C" {
+extern EFI_PEI_READ_ONLY_VARIABLE2_PPI  *PPIReadOnlyVariableServices;
+}
+
+#endif
```
Comment (os-d) [inline MdePkg/Test/Mock/Include/GoogleTest/Ppi/MockReadOnlyVariable2.h]: nit: need extra line at EOF

File: MdePkg/Test/Mock/Library/GoogleTest/Ppi/MockReadOnlyVariable2.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// Normally PPIVariableServices is "found"
+// This will be defined INSIDE the test, with its definition pointing to the mock function getVariable
+EFI_PEI_READ_ONLY_VARIABLE2_PPI  peiReadOnlyVariablePpi = {
+  GetVariable,      // EFI_PEI_GET_VARIABLE2
+  NextVariableName  // EFI_PEI_GET_NEXT_VARIABLE_NAME2
+};
+
+extern "C" {
+EFI_PEI_READ_ONLY_VARIABLE2_PPI  *PPIReadOnlyVariableServices = &peiReadOnlyVariablePpi;
+}
```
Comment (os-d) [inline MdePkg/Test/Mock/Library/GoogleTest/Ppi/MockReadOnlyVariable2.cpp]: nit: need extra line at EOF

**Review by os-d — APPROVED** [review]
This LGTM and I know we've had the conversation about how this will get consumed, but I forget. It is probably worthwhile to add a section in the readme on generic PPI/Protocol mocking for googletest and include how this should be consumed, as an example.

**Review by apop5 — APPROVED** [review]

File: MdePkg/Test/Mock/Include/GoogleTest/Ppi/MockReadOnlyVariable2.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/** @file MockReadOnlyVariable2.h
+  This file declares a mock of Read-only Variable Service2 PPI.
+  This ppi permits read-only access to the UEFI variable store during the PEI phase.
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
```
Comment (makubacki) [inline MdePkg/Test/Mock/Include/GoogleTest/Ppi/MockReadOnlyVariable2.h]: A little easier to read.

```suggestion
/** @file MockReadOnlyVariable2.h
  This file declares a mock of Read-only Variable Service2 PPI.
  
  This PPI permits read-only access to the UEFI variable store during the PEI phase.
  
  Copyright (c) Microsoft Corporation.
  SPDX-License-Identifier: BSD-2-Clause-Patent
**/

```

File: MdePkg/Test/Mock/Library/GoogleTest/Ppi/MockReadOnlyVariable2.cpp [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/** @file MockReadOnlyVariable2.cpp
+  Google Test mock for ReadOnlyVariable2
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
```
Comment (makubacki) [inline MdePkg/Test/Mock/Library/GoogleTest/Ppi/MockReadOnlyVariable2.cpp]:
```suggestion
/** @file MockReadOnlyVariable2.cpp
  Google Test mock for ReadOnlyVariable2
  
  Copyright (c) Microsoft Corporation.
  SPDX-License-Identifier: BSD-2-Clause-Patent
**/

```

File: MdePkg/Test/Mock/Library/GoogleTest/Ppi/MockReadOnlyVariable2.cpp [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+// Normally PPIVariableServices is "found"
+// This will be defined INSIDE the test, with its definition pointing to the mock function getVariable
+EFI_PEI_READ_ONLY_VARIABLE2_PPI  peiReadOnlyVariablePpi = {
+  GetVariable,      // EFI_PEI_GET_VARIABLE2
+  NextVariableName  // EFI_PEI_GET_NEXT_VARIABLE_NAME2
+};
+
+extern "C" {
+EFI_PEI_READ_ONLY_VARIABLE2_PPI  *PPIReadOnlyVariableServices = &peiReadOnlyVariablePpi;
```
Comment (makubacki) [inline MdePkg/Test/Mock/Library/GoogleTest/Ppi/MockReadOnlyVariable2.cpp]: Is there a reason these are using different casing?

For Pascal case, they'd be `PpiReadOnlyVariableServices` and `PeiReadOnlyVariablePpi`.

**Review by makubacki — APPROVED** [review]
Approved with a few suggestions.

**Pull request merged**